### ISSUE TITLE
mc mirror --exclude seems not to be working on windows

### ISSUE THREAD

**Issue by CrashLaker** [minio/mc#4082]
## Expected behavior
excluding folder when uploading
actually idk if it's actually only listing then excluding or both

## Actual behavior
not excluding

## Steps to reproduce the behavior
```
C:\Users\User>mc.exe mirror --exclude "AppData\*"  . windows-pc/backup/
 0 B / ? [                                                                                                           =]
mc.exe: <ERROR> Failed to perform mirroring Insufficient permissions to access this file `C:\Users\User\AppData\Local\ElevatedDiagnostics`
```

## mc --version
- (paste output of `mc --version`)
mc version RELEASE.2022-05-04T06-07-55Z

## System information
windows 10

**Comment by thebetauser** [comment]
Is there any update on this? Seems the --exclude only works on files however when backing up some python projects I want to exclude all the directories that are not needed such as .git and __pycache__ and the exclude commands doesn't work on folders

This command doesn't filter any of the folders

`mc mirror --exclude ".*" --exclude "venv" --exclude "*pycache*" --exclude "*-info" "E:\Python Projects" s3/work_backup`

MC Version:
`mc version RELEASE.2022-08-23T05-45-20Z (commit-id=07fffc3bc88f70a73c3714d9af23122f611b76cf)
Runtime: go1.18.5 windows/amd64
Copyright (c) 2015-2022 MinIO, Inc.
License GNU AGPLv3 <https://www.gnu.org/licenses/agpl-3.0.html>`

**Comment by cwinters8** [comment]
I can confirm this is still an issue. I'm unable to specify a directory to exclude using either the `dir` or `dir/*` syntax.

![exclude tmp from mirror](https://github.com/minio/mc/assets/40615752/652004b3-856e-4f57-80fc-73cb1c9ec1b3)

<img width="947" alt="exclude tmp/* from mirror" src="https://github.com/minio/mc/assets/40615752/23026d41-2372-4c34-97cb-c7284b509651">

**Comment by cwinters8** [comment]
Further experimentation seems to reveal `--exclude` is not functional with file names or specific paths to files either

<img width="938" alt="exclude hello.txt from mirror" src="https://github.com/minio/mc/assets/40615752/f45609e2-24ab-4faf-b5c9-b839a27c765b">

Also tried removing `--remove` and enclosing the path in quotes just in case those might be factors, but saw the same behavior

<img width="937" alt="exclude tmp/hello.txt using quotes" src="https://github.com/minio/mc/assets/40615752/8d11c8ee-e3ff-41e4-9753-3e905e3584e9">

FWIW I think the ability to parse an ignore file would be the most useful for excluding files. For example if `.mcignore` exists in the source, exclude all matching files using the same rules as git follows for `.gitignore`.

`mc` version:

```sh
mc version RELEASE.2024-01-05T05-04-32Z (commit-id=59eca9fea8984adec1e8e7a1c95d0ea23107ceff)
Runtime: go1.21.5 darwin/amd64
Copyright (c) 2015-2024 MinIO, Inc.
License GNU AGPLv3 <https://www.gnu.org/licenses/agpl-3.0.html>
```

**Comment by jiuker** [comment]
fix via #4825
